src/main/java/Objective.java

```diff
@@ -0,0 +1,23 @@
+import java.util.*;
+
+public class Objective {
+    private TimeLimit goal;
```

Comment:
I assume you are still working on this class, as some of these members are not used anywhere in the class nor can they be accessed from outside the class. If not then you should do something about them.

Good idea to break up the problem to such small pieces (classes).